Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp

```diff
@@ -145,199 +354,94 @@ namespace AzToolsFramework
-    bool LocalViewBookmarkLoader::ReadViewBookmarksFromSettingsRegistry()
-    {
-        struct ViewBookmarkVisitor : AZ::SettingsRegistryInterface::Visitor
+        const LocalViewBookmarkComponent* bookmarkComponent = FindOrCreateLocalViewBookmarkComponent();
+        if (!bookmarkComponent)
         {
-            ViewBookmarkVisitor()
-                : m_viewBookmarksKey{ ViewBookmarksRegistryPath } {};
-
-            AZ::SettingsRegistryInterface::VisitResponse Traverse(
-                AZStd::string_view path,
-                AZStd::string_view,
-                AZ::SettingsRegistryInterface::VisitAction action,
-                AZ::SettingsRegistryInterface::Type) override
-            {
-                if (action == AZ::SettingsRegistryInterface::VisitAction::Begin)
-                {
-                    // strip off the last json pointer key from the path and if it matches the view bookmark key then add an entry
-                    // to the view bookmark map
-                    AZStd::optional<AZStd::string_view> localBookmarksID = AZ::StringFunc::TokenizeLast(path, "/");
-                    if (path == m_viewBookmarksKey && localBookmarksID && !localBookmarksID->empty())
-                    {
-                        AZStd::string_view bookmarkKey = localBookmarksID.value();
-
-                        if (auto existingBookmarkEntry = m_bookmarkMap.find(bookmarkKey) == m_bookmarkMap.end())
-                        {
-                            m_bookmarkMap.insert(AZStd::make_pair(bookmarkKey, AZStd::vector<ViewBookmark>{}));
-                        }
-                    }
-                }
+            AZ_Assert(bookmarkComponent, "Couldn't find or create LocalViewBookmarkComponent.");
+            return false;
+        }
 
-                return AZ::SettingsRegistryInterface::VisitResponse::Continue;
-            }
+        const auto localBookmarksFileName = bookmarkComponent->GetLocalBookmarksFileName();
 
-            using AZ::SettingsRegistryInterface::Visitor::Visit;
-            void Visit(AZStd::string_view path, AZStd::string_view valueIndex, AZ::SettingsRegistryInterface::Type, double value) override
-            {
-                AZ::StringFunc::TokenizeLast(path, "/");
-                AZStd::optional<AZStd::string_view> dataType = AZ::StringFunc::TokenizeLast(path, "/");
-                AZStd::optional<AZStd::string_view> bookmarkIndexStr = AZ::StringFunc::TokenizeLast(path, "/");
-                AZStd::optional<AZStd::string_view> bookmarkType;
-                if (bookmarkIndexStr == LastKnownLocationKey)
-                {
-                    bookmarkType = bookmarkIndexStr;
-                }
-                else
-                {
-                    // differentiate between local Bookmarks and LastKnownLocation
-                    bookmarkType = AZ::StringFunc::TokenizeLast(path, "/");
-                }
+        bool success = false;
+        if (auto registry = AZ::SettingsRegistry::Get())
```

Comment:
you have this nested in an if statement but below its assigned to a variable and checked with a separate warning. 

```
 auto registry = AZ::SettingsRegistry::Get();
if (!registry)
{
AZ_Warning("LocalViewBookmarkLoader", false, "Unable to access global settings registry. Editor Preferences cannot be saved");
            return false;
}
```

Reply:
I'm not sure I follow sorry? Which function are you talking about?

Reply:
just structure how the registry is retrieved as this method: WriteViewBookmarksSettingsRegistryToFile.  just to make it consistent in the file probably not necessary though.

Reply:
Ah yes I see what you mean, I can update this later today :+1: Good suggestion cheers!